### PR TITLE
Portuguese Translation 2019 - Resource Hints Chapter

### PR DESCRIPTION
**Pull request by HakaCode**
Makes progress on #505

### CODE REVIEW

**Review by tunetheweb — APPROVED** [review]
LGTM!

Any of the @HTTPArchive/portuguese-translators want to review? If I don't hear anything I'll merge in next day or two.

File: src/content/pt/2019/ecommerce.md [hunk 1]
```diff
@@ -0,0 +1,639 @@
...
+
+Os proprietários de sites desejam que seus sites tenham uma boa aparência em dispositivos modernos. Como resultado, muitos sites oferecem as mesmas imagens de produtos de alta resolução para todos os usuários, _ndependentemente da resolução ou do tamanho da tela_. Os desenvolvedores podem não estar cientes (ou não querer usar) técnicas responsivas que permitem a entrega eficiente da melhor imagem possível para diferentes usuários. Vale lembrar que imagens de alta resolução podem não necessariamente aumentar as taxas de conversão. Por outro lado, o uso excessivo de imagens pesadas pode afetar a velocidade da página e, portanto, _reduzir_ as taxas de conversão. Na experiência dos autores em análises e eventos de sites, alguns desenvolvedores e outras partes interessadas têm SEO ou outras preocupações sobre o uso de carregamento lento para imagens.
```
Comment (emanuelgsouza) [inline src/content/pt/2019/ecommerce.md:460]: I think in the "_ndependentemente da resolução ou do tamanho da tela_" phrase is missing the `i` letter.

I believe the phrase "ou não querer usar" could be more fluid with "ou não querem usar"

File: src/content/pt/2019/ecommerce.md [hunk 2]
```diff
@@ -0,0 +1,639 @@
...
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vROHKGuMutXmxtzEoBSJNVn7DOzlfpizJh7mOkopFK8OVl_hCUHDOmKgYOojrpVsGnGWaletE7Uc5oX/pubchart?oid=2108999644&format=interactive"
+  )
+}}
+
+<p class="note">Observe que alguns serviços de imagem ou CDNs entregarão automaticamente WebP (em vez de JPEG ou PNG) para plataformas que suportam WebP, mesmo para uma URL com sufixo `.jpg` ou `.png`. Por exemplo, <a hreflang="en" href="https://res.cloudinary.com/webdotdev/f_auto/w_500/IMG_20190113_113201.jpg">IMG_20190113_113201.jpg</a> retorna uma imagem WebP no Chrome. No entanto, a maneira como o HTTP Archive <a hreflang="en" href="https://github.com/HTTPArchive/legacy.httparchive.org/blob/31a25b9064a365d746d4811a1d6dda516c0e4985/bulktest/batch_lib.inc#L994">é verificar primeiro as palavras-chave no tipo MIME e, em seguida, recorrer à extensão do arquivo. Isso significa que o formato para imagens com URLs como o acima será fornecido como WebP, uma vez que WebP é suportado pelo HTTP Archive como um agente de usuário.</p>
```
Comment (emanuelgsouza) [inline src/content/pt/2019/ecommerce.md:474]: It is missing the close `<a>` tag

File: src/content/pt/2019/ecommerce.md [hunk 3]
```diff
@@ -0,0 +1,639 @@
...
+
+[O First Contentful Paint](./performance#first-contentful-paint) mede o tempo que leva desde a navegação até que o conteúdo, como texto ou imagem, seja exibido pela primeira vez. Neste contexto, **rápido** significa FCP em menos de um segundo, **lento** significa FCP em 3 segundos ou mais e **moderado** é tudo o que está entre os dois. Observe que o conteúdo e o código de terceiros podem ter um impacto significativo no FCP.
+
+Todas as seis principais plataformas de comércio eletrônico têm pior FCP no celular do que no desktop: menos rápido e mais lento. Observe que o FCP é afetado pela capacidade do dispositivo (capacidade de processamento, memória, etc.), bem como pela conectividade.
+
+Precisamos estabelecer por que o FCP é pior no celular do que no desktop. Quais são as causas: conectividade e/ou capacidade do dispositivo ou outra coisa?
+
+## Pontuações do Progressive Web App (PWA)
+
+Consulte também o [capítulo PWA](./pwa) para obter mais informações sobre este tópico, além de apenas sites de comércio eletrônico.
```
Comment (emanuelgsouza) [inline src/content/pt/2019/ecommerce.md:619]: I think could be more fluid with "Capítulo sobre PWA" here.

File: src/content/pt/2019/ecommerce.md [hunk 4]
```diff
@@ -0,0 +1,639 @@
...
+  )
+}}
+
+Mais de 60% das páginas iniciais em plataformas de comércio eletrônico obtêm uma <a hreflang="en" href="https://developers.google.com/web/ilt/pwa/lighthouse-pwa-analysis-tool">pontuação Lighthouse PWA</a> entre 0.25 e 0.35. Menos de 20% das páginas iniciais em plataformas de comércio eletrônico obtêm uma pontuação de mais de 0.5 e menos de 1% das páginas iniciais obtêm uma pontuação superior a 0.6.
+
+O Lighthouse retorna uma pontuação do Progressive Web App (PWA) entre 0 e 1. 0 é a pior pontuação possível e 1 é a melhor. As auditorias do PWA são baseadas na <a hreflang="en" href="https://developers.google.com/web/progressive-web-apps/checklist"> Lista de Verificação de Base do PWA</a>, which lists 14 requirements. que lista 14 requisitos. O Lighthouse automatizou auditorias para 11 dos 14 requisitos. Os 3 restantes só podem ser testados manualmente. Cada uma das 11 auditorias PWA automatizadas são ponderadas igualmente, de modo que cada uma contribui com aproximadamente 9 pontos para sua pontuação PWA.
```
Comment (emanuelgsouza) [inline src/content/pt/2019/ecommerce.md:631]: There is a non-translated phrase here ("which lists 14 requirements")

**Review by emanuelgsouza — CHANGES_REQUESTED** [review]
Excellent work. Just a few considerations about the translate.

File: src/content/pt/2019/resource-hints.md [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+featured_stat_label_1: Sites using `dns-prefetch`
+featured_stat_2: 88%
+featured_stat_label_2: Resource hints using the `as` attribute
+featured_stat_3: 0.04%
+featured_stat_label_3: Usage of priority hints
+---
+
+## Introdução
+
+<a hreflang="en" href="https://www.w3.org/TR/resource-hints/">As dicas de recursos</a> fornecem "dicas" ao navegador sobre quais recursos serão necessários em breve. A ação que o navegador executa ao receber essa dica irá variar dependendo do tipo de dica de recurso; diferentes dicas de recursos dão início a diferentes ações. Quando usados ​​corretamente, eles podem melhorar o desempenho da página, dando uma vantagem inicial para ações antecipadas importantes.
```
[the remark this answers is not in the document]
Comment (HakaCode) [reply src/content/pt/2019/resource-hints.md:24]: > I believe this following phrase "dando uma vantagem inicial para ações antecipadas importantes.", could be more fluid with this "dando uma vantagem inicial para importantes ações antecipadas."

Very good @emanuelgsouza, I liked that, Thanks!

File: src/content/pt/2019/ecommerce.md (hunk 3 above)
Comment (HakaCode) [reply src/content/pt/2019/ecommerce.md:619]: Thanks

File: src/content/pt/2019/ecommerce.md (hunk 1 above)
Comment (HakaCode) [reply src/content/pt/2019/ecommerce.md:460]: > I think in the "_ndependentemente da resolução ou do tamanho da tela_" phrase is missing the `i` letter.
> 
> I believe the phrase "ou não querer usar" could be more fluid with "ou não querem usar"

Thanks

File: src/content/pt/2019/ecommerce.md (hunk 2 above)
Comment (HakaCode) [reply src/content/pt/2019/ecommerce.md:474]: Ok

File: src/content/pt/2019/ecommerce.md (hunk 4 above)
Comment (HakaCode) [reply src/content/pt/2019/ecommerce.md:631]: > There is a non-translated phrase here ("which lists 14 requirements")

fixed up

**Review by HakaCode — COMMENTED** [review]
suggestion fixed.

**Review by HakaCode — COMMENTED** [review]
all modifications fixed.
Allowed to merge!

File: src/content/pt/2019/resource-hints.md [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+discuss: 1774
+results: https://docs.google.com/spreadsheets/d/14QBP8XGkMRfWRBbWsoHm6oDVPkYhAIIpfxRn4iOkbUU/
+khempenius_bio: Katie Hempenius é engenheira da equipe do Chrome, onde trabalha para tornar a web mais rápida.
+featured_quote: As dicas de recursos fornecem <em>dicas</em> ao navegador sobre quais recursos serão necessários em breve. A ação que o navegador executa ao receber essa dica irá variar dependendo do tipo de dica de recurso; diferentes dicas de recursos dão início a diferentes ações. Quando usados corretamente, eles podem melhorar o desempenho da página, dando uma vantagem inicial para importantes ações antecipadas.
+featured_stat_1: 29%
+featured_stat_label_1: Sites using `dns-prefetch`
+featured_stat_2: 88%
+featured_stat_label_2: Resource hints using the `as` attribute
+featured_stat_3: 0.04%
+featured_stat_label_3: Usage of priority hints
```
Comment (tunetheweb) [inline src/content/pt/2019/resource-hints.md:19]: Looks like you've missed translating these

File: src/content/pt/2019/ecommerce.md [hunk 7]
```diff
@@ -0,0 +1,639 @@
...
+Das 116 plataformas de comércio eletrônico detectadas, apenas seis são encontradas em mais de 0,1% dos sites de desktop ou móveis. Observe que esses resultados não mostram variação por país, tamanho do site ou outras métricas semelhantes.
+
+A Figura 13.3 acima mostra que o WooCommerce tem a maior adoção em cerca de 4% dos sites de desktop e móveis. O Shopify é o segundo com cerca de 1,6% de adoção. Magento, PrestaShop, Bigcommerce e Shopware seguem com adoção cada vez menor, aproximando-se de 0,1%.
+
+### Cauda longa
+
+{{ figure_markup(
+  image="fig4.png",
+  caption="Adoção das principais plataformas de comércio eletrônico.",
+  description="Bar chart of the adoption of top 20 ecommerce platforms. Refer to Figure 13.3 above for a data table of adoption of the top six platforms.",
```
Comment (tunetheweb) [inline src/content/pt/2019/ecommerce.md:134]: Missed translation

File: src/content/pt/2019/ecommerce.md [hunk 8]
```diff
@@ -0,0 +1,639 @@
...
+  data_height=414
+  )
+}}
+
+Existem 110 plataformas de comércio eletrônico, cada uma com menos de 0,1% dos sites para desktop ou móveis. Cerca de 60 deles têm menos de 0,01% dos sites para celular ou desktop.
+
+{{ figure_markup(
+  image="fig5.png",
+  caption="Adoção combinada das seis principais plataformas de comércio eletrônico em comparação com as outras 110 plataformas.",
+  description="The top six ecommerce platforms make up 8% of all websites. The rest of the 110 platforms only make up 1.5% of websites. The results for desktop and mobile are similar.",
```
Comment (tunetheweb) [inline src/content/pt/2019/ecommerce.md:148]: Missed translation

File: src/content/pt/2019/ecommerce.md [hunk 9]
```diff
@@ -0,0 +1,639 @@
...
+7,87% de todas as solicitações no celular e 8,06% no desktop são para páginas iniciais em uma das seis principais plataformas de comércio eletrônico. Outros 1,52% das solicitações no celular e 1,59% no desktop são para páginas iniciais nas outras 110 plataformas de comércio eletrônico.
+
+## Comércio eletrônico (todas as plataformas)
+
+No total, 9,7% das páginas para desktop e 9,5% das páginas para celular usaram uma plataforma de comércio eletrônico.
+
+{{ figure_markup(
+  image="fig6.png",
+  caption="Porcentagem de páginas que usam qualquer plataforma de comércio eletrônico.",
+  description="9.7% of desktop pages use an ecommerce platform and 9.5% of mobile pages use an ecommerce platform.",
```
Comment (tunetheweb) [inline src/content/pt/2019/ecommerce.md:166]: Missed translation

File: src/content/pt/2019/ecommerce.md [hunk 10]
```diff
@@ -0,0 +1,639 @@
...
+Embora a proporção de sites de desktop seja ligeiramente maior no geral, algumas plataformas populares (incluindo WooCommerce, PrestaShop e Shopware) têm, na verdade, mais sites móveis do que sites de desktop.
+
+## Peso da página e solicitações
+
+O [peso da página](./page-weight) de uma plataforma de comércio eletrônico inclui todos os HTML, CSS, JavaScript, JSON, XML, imagens, áudio e vídeo.
+
+{{ figure_markup(
+  image="fig7.png",
+  caption="Distribuição do peso da página de comércio eletrônico.",
+  description="Distribution of the 10, 25, 50, 75, and 90th percentiles of ecommerce page weight. The median desktop ecommerce page loads 2.7 MB. The 10th percentile is 1.0 MB, 25th 1.6 MB, 75th 4.5 MB, and 90th 7.6 MB. Desktop websites are slightly higher than mobile by tenths of a megabyte.",
```
Comment (tunetheweb) [inline src/content/pt/2019/ecommerce.md:184]: Missed translation

File: src/content/pt/2019/ecommerce.md [hunk 11]
```diff
@@ -0,0 +1,639 @@
...
+  height=363,
+  data_width=600,
+  data_height=363
+  )
+}}
+
+{{ figure_markup(
+  image="fig8.png",
+  caption="Distribuição de solicitações por página de comércio eletrônico.",
+  description="Distribution of the 10, 25, 50, 75, and 90th percentiles of requests per ecommerce page. The median desktop ecommerce page makes 108 requests. The 10th percentile is 53 requests, 25th 76, 75th 153, and 90th 210. Desktop websites are slightly higher than mobile by about 10 requests.",
```
Comment (tunetheweb) [inline src/content/pt/2019/ecommerce.md:196]: Missed translation

File: src/content/pt/2019/ecommerce.md [hunk 12]
```diff
@@ -0,0 +1,639 @@
...
+
+## Tamanho da carga HTML
+
+{{ figure_markup(
+  image="fig11.png",
+  caption="Distribuição de bytes HTML (em KB) por página de comércio eletrônico.",
+  description="Distribution of the 10, 25, 50, 75, and 90th percentiles of HTML bytes per ecommerce page. The median desktop ecommerce page has 36 KB of HTML. The 10th percentile is 12 KB, 25th 20, 75th 66, and 90th 118. Desktop websites have slightly more HTML bytes than mobile by 1 or 2 KB.",
```
Comment (tunetheweb) [inline src/content/pt/2019/ecommerce.md:420]: Missed translation

File: src/content/pt/2019/ecommerce.md [hunk 13]
```diff
@@ -0,0 +1,639 @@
...
+  caption="Distribuição de bytes de imagem (em KB) por página de comércio eletrônico.",
+  description="Distribution of the 10, 25, 50, 75, and 90th percentiles of image bytes per ecommerce page. The median mobile ecommerce page has 1,517 KB of images. The 10th percentile is 318 KB, 25th 703, 75th 3,132, and 90th 5,881. Desktop and mobile websites have similar distributions.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vROHKGuMutXmxtzEoBSJNVn7DOzlfpizJh7mOkopFK8OVl_hCUHDOmKgYOojrpVsGnGWaletE7Uc5oX/pubchart?oid=323146848&format=interactive"
+  )
+}}
+
+{{ figure_markup(
+  image="fig13.png",
+  caption="Distribuição de solicitações de imagem por página de comércio eletrônico.",
+  description="Distribution of the 10, 25, 50, 75, and 90th percentiles of image requests per ecommerce page. The median desktop ecommerce page makes 40 image requests. The 10th percentile is 16 requests, 25th 25, 75th 62, and 90th 97. The desktop distribution is slightly higher than mobile by 2-10 requests at each percentile.",
```
Comment (tunetheweb) [inline src/content/pt/2019/ecommerce.md:442]: Missed translation

File: src/content/pt/2019/ecommerce.md [hunk 14]
```diff
@@ -0,0 +1,639 @@
...
+
+Precisamos fazer mais análises para entender melhor por que alguns sites não estão usando técnicas de imagem responsiva ou carregamento lento. Também precisamos fornecer orientação que ajude as plataformas de comércio eletrônico a entregar belas imagens de maneira confiável para aqueles com dispositivos de última geração e boa conectividade, ao mesmo tempo em que fornece a melhor experiência possível para dispositivos de baixo custo e com conectividade ruim.
+
+## Formatos de imagem populares
+
+{{ figure_markup(
+  image="fig15.png",
+  caption="Formatos de imagem populares.",
+  description="Bar chart showing the popularity of various image formats. JPEG is the most popular format at 54% of images on desktop ecommerce pages. Next are PNG at 27%, GIF at 14%, SVG at 2%, and WebP and ICO at 1% each.",
```
Comment (tunetheweb) [inline src/content/pt/2019/ecommerce.md:469]: Missed translation

File: src/content/pt/2019/ecommerce.md [hunk 15]
```diff
@@ -0,0 +1,639 @@
...
+
+Muitos sites - especialmente lojas online - carregam uma quantidade significativa de código e conteúdo de terceiros: para análises, testes A / B, rastreamento do comportamento do cliente, publicidade e suporte de mídia social. O conteúdo de terceiros pode ter um <a hreflang="en" href="https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/loading-third-party-javascript">impacto significativo no desempenho</a>. [Patrick Hulce](https://twitter.com/patrickhulce)'s <a hreflang="en" href="https://github.com/patrickhulce/third-party-web">third-party-web tool</a> é usado para determinar os pedidos de terceiros para este relatório, e isso é discutido mais no [Terceiros](./third-parties) capítulo.
+
+{{ figure_markup(
+  image="fig17.png",
+  caption="Distribuição de solicitações de terceiros por página de comércio eletrônico.",
+  description="Distribution of the 10, 25, 50, 75, and 90th percentiles of third-party requests per ecommerce page. The median number of third-party requests on desktop is 19. The 10, 25, 75, and 90th percentiles are: 4, 9, 34, and 54. The desktop distribution is higher at each percentile than mobile by only 1-2 requests.",
```
Comment (tunetheweb) [inline src/content/pt/2019/ecommerce.md:560]: Missed translation

File: src/content/pt/2019/ecommerce.md [hunk 16]
```diff
@@ -0,0 +1,639 @@
...
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vROHKGuMutXmxtzEoBSJNVn7DOzlfpizJh7mOkopFK8OVl_hCUHDOmKgYOojrpVsGnGWaletE7Uc5oX/pubchart?oid=865791628&format=interactive"
+  )
+}}
+
+{{ figure_markup(
+  image="fig18.png",
+  caption="Distribuição de bytes de terceiros por página de comércio eletrônico.",
+  description="Distribution of the 10, 25, 50, 75, and 90th percentiles of third-party bytes per ecommerce page. The median number of third-party bytes on desktop is 320 KB. The 10, 25, 75, and 90th percentiles are: 42, 129, 651, and 1,071. The desktop distribution is higher at each percentile than mobile by 10-30 KB.",
```
Comment (tunetheweb) [inline src/content/pt/2019/ecommerce.md:568]: Missed translation

File: src/content/pt/2019/ecommerce.md [hunk 17]
```diff
@@ -0,0 +1,639 @@
...
+<a hreflang="en" href="https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/loading-third-party-javascript/">Outros estudos</a> indicaram que o conteúdo de terceiros pode ser um grande gargalo de desempenho. Este estudo mostra que 17 ou mais solicitações (50 ou mais para os 10% principais) é a norma para páginas de comércio eletrônico.
+
+## Solicitações de terceiros e carga útil por plataforma
+
+Observe que os gráficos e tabelas abaixo mostram dados apenas para dispositivos móveis.
+
+{{ figure_markup(
+  image="fig19.png",
+  caption="Distribuição de solicitações de terceiros por página móvel para cada plataforma de comércio eletrônico.",
+  description="Distribution of the 10, 25, 50, 75, and 90th percentiles of third-party requests per ecommerce page for each platform. Shopify and Bigcommerce load the most third-party requests across the distributions by about 40 requests at the median.",
```
Comment (tunetheweb) [inline src/content/pt/2019/ecommerce.md:584]: Missed translation

File: src/content/pt/2019/ecommerce.md [hunk 18]
```diff
@@ -0,0 +1,639 @@
...
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vROHKGuMutXmxtzEoBSJNVn7DOzlfpizJh7mOkopFK8OVl_hCUHDOmKgYOojrpVsGnGWaletE7Uc5oX/pubchart?oid=1242665725&format=interactive"
+  )
+}}
+
+{{ figure_markup(
+  image="fig20.png",
+  caption="Distribuição de bytes de terceiros (KB) por página móvel para cada plataforma de comércio eletrônico.",
+  description="Distribution of the 10, 25, 50, 75, and 90th percentiles of third-party bytes (KB) per ecommerce page for each platform. Shopify and Bigcommerce load the most third-party bytes across the distributions by more than 1,000 KB at the median.",
```
Comment (tunetheweb) [inline src/content/pt/2019/ecommerce.md:592]: Missed translation

File: src/content/pt/2019/ecommerce.md [hunk 19]
```diff
@@ -0,0 +1,639 @@
...
+Plataformas como Shopify podem estender seus serviços usando JavaScript do lado do cliente, enquanto outras plataformas como Magento usam mais extensões do lado do servidor. Essa diferença na arquitetura afeta as figuras vistas aqui.
+
+Claramente, as páginas em algumas plataformas de comércio eletrônico fazem mais solicitações de conteúdo de terceiros e geram uma carga útil maior de conteúdo de terceiros. Uma análise mais aprofundada pode ser feita sobre o _motivo_ pelo qual as páginas de algumas plataformas fazem mais solicitações e têm cargas úteis de terceiros maiores do que outras.
+
+## Primeira pintura com conteúdo (FCP)
+
+{{ figure_markup(
+  image="fig21.png",
+  caption="Distribuição média de experiências FCP por plataforma de comércio eletrônico.",
+  description="Bar chart of the average distribution of FCP experiences for the top six ecommerce platforms. WooCommerce has the highest density of slow FCP experiences at 43%. Shopify has the highest density of fast FCP experiences at 47%.",
```
Comment (tunetheweb) [inline src/content/pt/2019/ecommerce.md:606]: Missed translation

File: src/content/pt/2019/ecommerce.md [hunk 20]
```diff
@@ -0,0 +1,639 @@
...
+Precisamos estabelecer por que o FCP é pior no celular do que no desktop. Quais são as causas: conectividade e/ou capacidade do dispositivo ou outra coisa?
+
+## Pontuações do Progressive Web App (PWA)
+
+Consulte também o [Capítulo sobre PWA](./pwa) para obter mais informações sobre este tópico, além de apenas sites de comércio eletrônico.
+
+{{ figure_markup(
+  image="fig22.png",
+  caption="Distribution of Lighthouse PWA category scores for mobile ecommerce pages.",
+  description="Distribution of Lighthouse's PWA category scores for ecommerce pages. On a scale of 0 (failing) to 1 (perfect), 40% of pages get a score of 0.33. 1% of pages get a score above 0.6.",
```
Comment (tunetheweb) [inline src/content/pt/2019/ecommerce.md:624]: Missed translation

File: src/content/pt/2019/ecommerce.md [hunk 21]
```diff
@@ -0,0 +1,639 @@
...
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vROHKGuMutXmxtzEoBSJNVn7DOzlfpizJh7mOkopFK8OVl_hCUHDOmKgYOojrpVsGnGWaletE7Uc5oX/pubchart?oid=2108999644&format=interactive"
+  )
+}}
+
+<p class="note">Observe que alguns serviços de imagem ou CDNs entregarão automaticamente WebP (em vez de JPEG ou PNG) para plataformas que suportam WebP, mesmo para uma URL com sufixo `.jpg` ou `.png`. Por exemplo, <a hreflang="en" href="https://res.cloudinary.com/webdotdev/f_auto/w_500/IMG_20190113_113201.jpg">IMG_20190113_113201.jpg</a> retorna uma imagem WebP no Chrome. No entanto, a maneira como o HTTP Archive <a hreflang="en" href="https://github.com/HTTPArchive/legacy.httparchive.org/blob/31a25b9064a365d746d4811a1d6dda516c0e4985/bulktest/batch_lib.inc#L994"></a> é verificar primeiro as palavras-chave no tipo MIME e, em seguida, recorrer à extensão do arquivo. Isso significa que o formato para imagens com URLs como o acima será fornecido como WebP, uma vez que WebP é suportado pelo HTTP Archive como um agente de usuário.</p>
```
Comment (tunetheweb) [inline src/content/pt/2019/ecommerce.md:474]: This is causing your Test to fail. We have a blank link.

I think you've missed the phrase "detects image format as the original text is:

> However, the way HTTP Archive `<a hreflang="en" href="https://github.com/HTTPArchive/legacy.httparchive.org/blob/31a25b9064a365d746d4811a1d6dda516c0e4985/bulktest/batch_lib.inc#L994">`detects image formats`</a>` is to check for keywords in the MIME type first

**Review by tunetheweb — CHANGES_REQUESTED** [review]
Hi @HakaCode - thanks for your work on this but we also need all the tests to pass before we can merge. I've spotted the reason it's failing and a few other issues for you to look at.

Also in src/templates/pt/base.html we have the following at lines 159-161 which this is now highlighting as an error:

```json
    "ecommerce": "Comércio eletrônico",
    "cms": "CMS",
    "compression": "Comércio eletrônico",
```

Can we correct the `compression` chapter name to `Compressão`?

Also in general it's best to open a new branch and a new Pull Request for each translation rather than adding a new chunk of work to an existing pull request. That way we can merge what is ready to go, and hold back what isn't, rather than having to do both together.

File: src/content/pt/2019/resource-hints.md [hunk 22]
```diff
@@ -228,15 +228,15 @@ Embora outros elementos HTML suportem o atributo crossorigin, esta análise apen
...
+88% das instâncias de dicas de recursos usam o atributo `as`. Quando `as` é especificado, é predominantemente usado para scripts: 92% do uso é para scripts, 3% para fontes e 3% para estilos. Isso não é surpreendente, dado o papel proeminente que os scripts desempenham na arquitetura da maioria dos sites, bem como a alta frequência com que os scripts são usados ​​como vetores de ataque (tornando, portanto, particularmente importante que os scripts obtenham o CSP correto aplicado a eles).
 
 ## O futuro
 
 No momento, não há propostas para expandir o conjunto atual de dicas de recursos. No entanto, dicas de prioridade e carregamento lento nativo são duas tecnologias propostas que são semelhantes em espírito às dicas de recursos, pois fornecem APIs para otimizar o processo de carregamento.
 
-### Dicas de prioridade
+### Sugestões prioritárias
```
Comment (emanuelgsouza) [inline src/content/pt/2019/resource-hints.md:237]: I think here the best translation is "Dicas de prioridade". What do you think?

**Review by emanuelgsouza — CHANGES_REQUESTED** [review]
Great work, just a last consideration.

File: src/content/pt/2019/resource-hints.md (hunk 22 above)
Comment (HakaCode) [reply src/content/pt/2019/resource-hints.md:237]: > I think here the best translation is "Dicas de prioridade". What do you think?

@emanuelgsouza, would also be very nice, we can substitute if you want.

Comment (emanuelgsouza) [reply src/content/pt/2019/resource-hints.md:237]: Great, so if you can, please change it. Thanks

**Review by HakaCode — COMMENTED** [review]
fix-up

**Comment by tunetheweb** [comment]
> fix-up

I still see a number of untranslated `description=` lines in the ecommerce.md file.

**Comment by HakaCode** [comment]
> > consertar
> 
> Ainda vejo várias `description=`linhas não traduzidas no arquivo ecommerce.md.

@tunetheweb sorry, as i haven't used github in a while i'm a bit lost

**Comment by tunetheweb** [comment]
Easiest thing is to click on the Files Changed tab, and Load the full differences:

image

Then scroll down for all the unresolved issues.

Alternatively, because these are all issues with the `description=` lines you can just search for that in your files outside of GitHub and hopefully you'll see what I mean.

**Review by HakaCode — COMMENTED** [review]
I finally managed to locate myself and changes (as far as I can see) were completed, in case something is missing let me know :relieved:

File: src/content/pt/2019/ecommerce.md [hunk 23]
```diff
@@ -0,0 +1,639 @@
...
+
+Os tamanhos da carga útil do HTML móvel não são muito diferentes dos do desktop. Em outras palavras, parece que os sites não estão entregando arquivos HTML significativamente diferentes para dispositivos ou tamanhos de janela de visualização diferentes. Em muitos sites de comércio eletrônico, as cargas úteis de HTML da página inicial são grandes. Não sabemos se isso se deve a um HTML inchado ou a outro código (como JSON) nos arquivos HTML.
+
+## Estatísticas de imagem
+
+{{ figure_markup(
+  image="fig12.png",
+  caption="Distribuição de bytes de imagem (em KB) por página de comércio eletrônico.",
+  description="Distribution of the 10, 25, 50, 75, and 90th percentiles of image bytes per ecommerce page. The median mobile ecommerce page has 1,517 KB of images. The 10th percentile is 318 KB, 25th 703, 75th 3,132, and 90th 5,881. Desktop and mobile websites have similar distributions.",
```
Comment (tunetheweb) [inline src/content/pt/2019/ecommerce.md:434]: Can you look at this one @HakaCode ?

**Review by tunetheweb — CHANGES_REQUESTED** [review]
Almost there - just one more to go!

**Review by tunetheweb — APPROVED** [review]

**Comment by tunetheweb** [comment]
Thanks @HakaCode and @emanuelgsouza - looks great now!

**Pull request merged**